addons/http_routing/models/ir_http.py

```diff
@@ -661,8 +663,7 @@ def url_rewrite(self, path, query_args=None):
         except werkzeug.exceptions.MethodNotAllowed:
             endpoint = router.match(path, method='GET', query_args=query_args)
         except werkzeug.routing.RequestRedirect as e:
-            # get path from http://{path}?{current query string}
-            new_url = e.new_url.split('?')[0][7:]
+            new_url = e.new_url.split('?')[0][7:]  # remove scheme
```

Comment:
It is a revert of commit odoo/odoo@4e24115
that seems a first view wrong since the QS was moved here: https://github.com/odoo/odoo/commit/3beb4545c4599469eff77a247db12b630cace754#diff-10359953e238a564cd8fb556ca799d7068e6ba77d9821404b1e9004e63b1426cR197

maybe @nle-odoo can confirm the need of this fix, and why the move doesnt work for his usecase?

Reply:
I made a test for that particular commit from NLE, on his PR, https://github.com/odoo/odoo/pull/81685/commits/5cf388449386ded12743ae40d85211b4b5b6fa24
Let me know if the test is somehow not catching something that is now broken, or not correctly testing in master what it tested in 14.0